code/modules/events/ghost_role/venom.dm

```diff
@@ -0,0 +1,484 @@
+	mob_size = MOB_SIZE_TINY
+	mob_biotypes = MOB_EPIC
+	see_in_dark = 8
+	sight = SEE_MOBS
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_plas" = 0, "max_plas" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	del_on_death = TRUE
+	var/obj/song_player
+	var/charging = FALSE
+	var/power = 0
+	var/datum/song/song
+	var/obj/item/mod/control/mod
+
+/mob/living/simple_animal/hostile/venom/Initialize(mapload)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+	song_player = new(src)
+	song = new(src, "meowsynth", 0)
+	song.max_repeats = INFINITY
+	song.repeat = INFINITY
+	song.volume = 50
+	song.ParseSong(VENOM_SONG)
+	INVOKE_ASYNC(song, TYPE_PROC_REF(/datum/song, start_playing), song_player)
+	var/datum/action/cooldown/telepathy/telepathy = new(src)
+	telepathy.Grant(src)
+	var/datum/action/cooldown/mind_control/control = new(src)
+	control.Grant(src)
+
+/mob/living/simple_animal/hostile/venom/death()
+	if(!QDELETED(mod))
+		qdel(mod)
+	return ..()
+
+/mob/living/simple_animal/hostile/venom/Destroy()
+	QDEL_NULL(song)
+	QDEL_NULL(song_player)
+	return ..()
+
+/mob/living/simple_animal/hostile/venom/get_status_tab_items()
+	. = ..()
+	. += "Power: [power]"
+
+/mob/living/simple_animal/hostile/venom/UnarmedAttack(atom/attack_target, proximity_flag, list/modifiers)
+	OpenFire(attack_target)
+
+/mob/living/simple_animal/hostile/venom/OpenFire(atom/target)
+	if(charging || mod)
+		return
+	visible_message(span_danger("<b>[src]</b> [ranged_message] at [target]!"))
+	COOLDOWN_START(src, ranged_cooldown, ranged_cooldown_time)
+	Shoot(target)
+
+/mob/living/simple_animal/hostile/venom/Shoot(atom/targeted_atom)
+	charging = TRUE
+	throw_at(targeted_atom, range = 5, speed = 1.5, thrower = src, callback = CALLBACK(src, PROC_REF(charging_end)))
+
+/mob/living/simple_animal/hostile/venom/ex_act(severity, target, origin)
+	return
+
+/mob/living/simple_animal/hostile/venom/proc/charging_end()
+	charging = FALSE
+
+/mob/living/simple_animal/hostile/venom/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	if(!charging)
+		return ..()
+	if(!ishuman(hit_atom))
+		return
+	var/mob/living/carbon/human/hit_mob = hit_atom
+	var/obj/item/mod/control/venom_target = locate() in hit_mob
+	if(!venom_target || !venom_target.active || venom_target.activating)
+		return
+	hit_mob.visible_message(span_danger("<b>[src]</b> jumps onto [hit_mob]!"), span_userdanger("<b>[src]</b> jumps onto you!"))
+	to_chat(hit_mob, span_hypnophrase("You feel like listening to it..."))
+	shake_camera(hit_mob, 4, 3)
+	shake_camera(src, 2, 3)
+	venomify_mod(venom_target)
+
+/mob/living/simple_animal/hostile/venom/proc/venomify_mod(obj/item/mod/control/mod_target)
+	mod = mod_target
+	RegisterSignal(mod, COMSIG_MOD_ACTIVATE, PROC_REF(block_mod_activation))
+	RegisterSignal(mod, COMSIG_PARENT_QDELETING, PROC_REF(death))
+	ADD_TRAIT(mod, TRAIT_NODROP, REF(src))
+	ADD_TRAIT(mod.wearer, TRAIT_NOHUNGER, REF(src))
```

Comment:
Curious, he claims it has no hunger, yet it has nohunger.

This says a lot about venom antag 2022.